File: api/panda-pay/src/user/user.controller.ts
import { Controller, Get, Post, Body, Res, Param } from '@nestjs/common';
import { UserService } from './user.service';
import { UserDTO } from 'src/DTO/userDTO';
import { Response } from 'express';
import { User } from 'src/entity/user.entity';
@Controller('user')
export class UserController {
    constructor(
        private userService: UserService
    ) { }

    @Post('signup')
    async create(@Body() user, @Res() res: Response) {
        try {
            var req = new User();
            req = await this.userService.create(user);
            if (req) {
                res.json(req);
            } else {
                res.json({ "message": "dont create" });
            }

        }
        catch (e) {
            console.log(e);
        }
    }

    @Get()
    async allUser(@Res() res: Response) {
        let user = new Array<User>();
        user = await this.userService.allUsers();
        if (user) {
            res.status(200).json(user);
        }

    }

    @Get(':id')
    async getUser(@Param('id') id, @Res() res: Response) {
        let user = new User();
        user = await this.userService.getUser(id);
        if (user) {
            res.status(200).json(user);
        }

    }

    @Get('user_cpf/:cpf')
    async isUser(@Param('cpf') phone, @Res() res: Response) {
        let user = new User();
        user = await this.userService.isUser(phone);
        if (user) {
            res.json({ "message": true });
        } else {
            res.json({ "message": false });
        }
    }

    @Post('login')
    async login(@Body() userDTO: UserDTO, @Res() res: Response) {
        let user = new User();
        user = await this.userService.login(userDTO.cpf, userDTO.password);

        if (user) {
            res.json(user);
        } else {
            res.json({ "message": "login invalid" });
        }
    }

    @Get('get_balance/:id')
    async getBalance(@Param('id') id: number, @Res() res: Response) {
        var user = new User();
        try {
            user = await this.userService.getBalance(id);

            if (user) {
                res.json({ "balance": user.balance })
            }
        }
        catch (e) {
            console.log(e);
        }

    }

}

File: app_pandapay/lib/controller/user_controller.dart
import 'package:app_pandapay/model/user.dart';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:dio/dio.dart';

class UserController {
  var host = 'http://192.168.0.11:8080/user';
  Dio dio = new Dio();
  Response response;
  SharedPreferences prefs;

  Future<bool> isUser(String cpf) async {
    try {
      response = await dio.get(host + '/user_cpf/${cpf}');
      return response.data['message'];
    } catch (e) {
      print(e);
    }
  }

  Future<bool> login(User user) async {
    try {
      response = await dio.post(host + '/login',
          data: {"cpf": user.cpf, "password": user.password});

      if (response.data['message'] == null) {
        User user = User.fromJson(response.data);
        auth(user.id);

        return true;
      } else {
        return false;
      }
    } catch (e) {
      print(e);
    }
  }

  Future<bool> create(User user) async {
    try {
      response = await dio.post(
        host + '/signup',
        data: user.toJson(),
      );
      if (response.data['message'] == null) {
        auth(response.data['id']);
        return true;
      } else {
        return false;
      }
    } catch (e) {
      print(e.data);
    }
  }

  Future auth(int id) async {
    prefs = await SharedPreferences.getInstance();
    prefs.setInt('idUser', id);
    prefs.setBool('isAuth', true);
  }

  Future<bool> isAuth() async {
    prefs = await SharedPreferences.getInstance();
    return prefs.getBool('isAuth');
  }

  Future<String> getBalance() async {
    prefs = await SharedPreferences.getInstance();
    var id = prefs.getInt('idUser');    try {
      response = await dio.get(host + '/get_balance/${id}');

      if (response.data != null) {
        return response.data['balance'];
      } else {
        print(response.data);
      }
    } catch (e) {
      print(e);
    }
  }

  Future<User> getUser() async {
    prefs = await SharedPreferences.getInstance();
    var id = prefs.getInt('idUser');
    try {
      response = await dio.get(host + '/${id}');
      if (response.data != null) {
        User user = User.fromJson(response.data);

        return user;
      } else {
        print(response.data);
      }
    } catch (e) {
      print(e.data);
    }
  }
}
